File: IpcSample/IpcSample.Shared/IpcClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using System.Runtime.Remoting;
using System.Runtime.Remoting.Channels;
using System.Runtime.Remoting.Channels.Ipc;

namespace IpcSample.Shared
{
    public class IpcClient
    {
        public IpcClient()
        {
        }

        private IpcData _ipcdata = null;

        public IpcData IpcData
        {
            get { return _ipcdata; }
        }

        public void open()
        {
            // クライアントチャンネルの生成
            IpcClientChannel channel = new IpcClientChannel();

            // チャンネル登録
            ChannelServices.RegisterChannel( channel, ensureSecurity: true );

            // IPCサーバで公開されているIPCオブジェクトを取得する
            // URLは"ipc://{チャンネル名}/{公開名}"
            _ipcdata = Activator.GetObject( typeof( IpcData ), url: $"ipc://{ IpcServer.Channel }/ipcdata" ) as IpcData;
        }
    }
}

File: IpcSample/IpcSample.CLRLib/src/ipcclient.cpp
#include "ipcclient.h"

using namespace System;
using namespace System::Runtime::Remoting;
using namespace System::Runtime::Remoting::Channels;
using namespace System::Runtime::Remoting::Channels::Ipc;

namespace IpcSample
{
namespace Shared
{

IpcClient::IpcClient()
    : m_ipcdata( gcnew IpcData() )
{
}

void IpcClient::open()
{
    IpcClientChannel^ channel = gcnew IpcClientChannel();

    ChannelServices::RegisterChannel( channel, true );
    
    m_ipcdata = (IpcData^)( Activator::GetObject( m_ipcdata->GetType(), "ipc://ipcchannel/ipcdata" ) );
}

} // namespace Shared
} // namespace IpcSample
